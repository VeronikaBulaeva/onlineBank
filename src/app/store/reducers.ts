import { combineReducers } from "@reduxjs/toolkit";
import creditSlice from "./slices/creditSlice.ts";

const rootReducer = combineReducers({
  credit: creditSlice,
});

export default rootReducer;
