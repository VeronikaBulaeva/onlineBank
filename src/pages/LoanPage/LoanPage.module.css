.section {
    padding-top: 24px;
    padding-bottom: 96px;
    margin-inline: auto;
    max-width: 1300px;
}
