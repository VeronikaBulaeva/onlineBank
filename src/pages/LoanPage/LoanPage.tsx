import { FC } from "react";
import ApplicationIdForm from "@/components/forms/ApplicationIdForm.tsx";
import styles from "./LoanPage.module.css";

const LoanPage: FC = () => {
  return (
    <div className={styles.section}>
      <ApplicationIdForm />
    </div>
  );
};

export default LoanPage;
