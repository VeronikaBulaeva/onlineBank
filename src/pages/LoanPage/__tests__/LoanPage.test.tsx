import { describe, expect, it } from "vitest";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { store } from "@/app/store/store.ts";
import { BrowserRouter as Router } from "react-router-dom";
import LoanPage from "../LoanPage.tsx";

describe("LoanPage", () => {
  it("renders correctly", async () => {
    const { container } = render(
      <Provider store={store}>
        <Router>
          <LoanPage />
        </Router>
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
