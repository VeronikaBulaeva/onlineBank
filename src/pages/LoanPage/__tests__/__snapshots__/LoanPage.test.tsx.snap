// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoanPage > renders correctly 1`] = `
<div>
  <div
    class="_section_8b32ce"
  >
    <div
      class="_wrapper_b45c30 false"
    >
      <form
        class="_form_b45c30"
        name="aplicationIdForm"
        novalidate=""
      >
        <div
          class="_form__top_b45c30"
        >
          <div
            class="_form__text_b45c30"
          >
            <p
              class="_form__title_b45c30"
            >
              Continuation of the application
            </p>
            <p
              class="_step__text_b45c30"
            >
              Step 2 of 5
            </p>
          </div>
        </div>
        <div
          class="_top__inputs_b45c30"
        >
          <div
            class="_input__block_8f3183"
          >
            <label
              class="_label_8f3183"
            >
              <p
                class="_required_8f3183"
              >
                What's your gender
              </p>
              <select
                class="_input_b45c30"
                data-testid="gender"
                id="gender"
                name="gender"
                required=""
              >
                <option
                  class="_option_8f3183"
                  value=""
                />
                <option
                  class="_option_8f3183"
                  value="MALE"
                >
                  MALE
                </option>
                <option
                  class="_option_8f3183"
                  value="FEMALE"
                >
                  FEMALE
                </option>
              </select>
            </label>
          </div>
          <div
            class="_input__block_8f3183"
          >
            <label
              class="_label_8f3183"
            >
              <p
                class="_required_8f3183"
              >
                Your marital status
              </p>
              <select
                class="_input_b45c30"
                data-testid="maritalStatus"
                id="maritalStatus"
                name="maritalStatus"
                required=""
              >
                <option
                  class="_option_8f3183"
                  value=""
                />
                <option
                  class="_option_8f3183"
                  value="MARRIED"
                >
                  MARRIED
                </option>
                <option
                  class="_option_8f3183"
                  value="DIVORCED"
                >
                  DIVORCED
                </option>
                <option
                  class="_option_8f3183"
                  value="SINGLE"
                >
                  SINGLE
                </option>
                <option
                  class="_option_8f3183"
                  value="WIDOW_WIDOWER"
                >
                  WIDOW WIDOWER
                </option>
              </select>
            </label>
          </div>
          <div
            class="_input__block_8f3183"
          >
            <label
              class="_label_8f3183"
            >
              <p
                class="_required_8f3183"
              >
                Your number of dependents
              </p>
              <input
                class="false _input_b45c30"
                data-testid="dependentAmount"
                id="input"
                name="dependentAmount"
                required=""
                type="number"
              />
            </label>
          </div>
          <div
            class="_input__block_8f3183"
          >
            <label
              class="_label_8f3183"
            >
              <p
                class="_required_8f3183"
              >
                Date of issue of the passport
              </p>
              <input
                class="undefined false _input_b45c30 _big_b45c30"
                data-testid="passportIssueDate"
                name="passportIssueDate"
                placeholder="Select Date and Time"
                required=""
                value=""
              />
            </label>
          </div>
          <div
            class="_input__block_8f3183"
          >
            <label
              class="_label_8f3183"
            >
              <p
                class="_required_8f3183"
              >
                Division code
              </p>
              <input
                class="undefined false _input_b45c30 _big_b45c30"
                data-testid="passportIssueBranch"
                name="passportIssueBranch"
                placeholder="000-000"
                required=""
                value=""
              />
            </label>
          </div>
        </div>
        <div
          class="_form__bottom_b45c30"
        >
          <p
            class="_block__title_b45c30"
          >
            Employment
          </p>
          <div
            class="_bottom__inputs_b45c30"
          >
            <div
              class="_input__block_8f3183"
            >
              <label
                class="_label_8f3183"
              >
                <p
                  class="_required_8f3183"
                >
                  Your employment status
                </p>
                <select
                  class="_input_b45c30"
                  data-testid="employmentStatus"
                  id="employmentStatus"
                  name="employmentStatus"
                  required=""
                >
                  <option
                    class="_option_8f3183"
                    value=""
                  />
                  <option
                    class="_option_8f3183"
                    value="UNEMPLOYED"
                  >
                    UNEMPLOYED
                  </option>
                  <option
                    class="_option_8f3183"
                    value="SELF_EMPLOYED"
                  >
                    SELF EMPLOYED
                  </option>
                  <option
                    class="_option_8f3183"
                    value="EMPLOYED"
                  >
                    EMPLOYED
                  </option>
                  <option
                    class="_option_8f3183"
                    value="BUSINESS_OWNER"
                  >
                    BUSINESS OWNER
                  </option>
                </select>
              </label>
            </div>
            <div
              class="_input__block_8f3183"
            >
              <label
                class="_label_8f3183"
              >
                <p
                  class="_required_8f3183"
                >
                  Your employer INN
                </p>
                <input
                  class="undefined false _input_b45c30"
                  data-testid="employerINN"
                  name="employerINN"
                  placeholder="000000000000"
                  required=""
                  value=""
                />
              </label>
            </div>
            <div
              class="_input__block_8f3183"
            >
              <label
                class="_label_8f3183"
              >
                <p
                  class="_required_8f3183"
                >
                  Your salary
                </p>
                <input
                  class="false _input_b45c30"
                  data-testid="salary"
                  id="input"
                  name="salary"
                  placeholder="For example 100 000"
                  required=""
                  type="number"
                />
              </label>
            </div>
            <div
              class="_input__block_8f3183"
            >
              <label
                class="_label_8f3183"
              >
                <p
                  class="_required_8f3183"
                >
                  Your position
                </p>
                <select
                  class="_input_b45c30"
                  data-testid="position"
                  id="position"
                  name="position"
                  required=""
                >
                  <option
                    class="_option_8f3183"
                    value=""
                  />
                  <option
                    class="_option_8f3183"
                    value="WORKER"
                  >
                    WORKER
                  </option>
                  <option
                    class="_option_8f3183"
                    value="MID_MANAGER"
                  >
                    MID MANAGER
                  </option>
                  <option
                    class="_option_8f3183"
                    value="TOP_MANAGER"
                  >
                    TOP MANAGER
                  </option>
                  <option
                    class="_option_8f3183"
                    value="OWNER"
                  >
                    OWNER
                  </option>
                </select>
              </label>
            </div>
            <div
              class="_input__block_8f3183"
            >
              <label
                class="_label_8f3183"
              >
                <p
                  class="_required_8f3183"
                >
                  Your work experience total
                </p>
                <input
                  class="false _input_b45c30"
                  data-testid="workExperienceTotal"
                  id="input"
                  name="workExperienceTotal"
                  placeholder="For example 10"
                  required=""
                  type="number"
                />
              </label>
            </div>
            <div
              class="_input__block_8f3183"
            >
              <label
                class="_label_8f3183"
              >
                <p
                  class="_required_8f3183"
                >
                  Your work experience current
                </p>
                <input
                  class="false _input_b45c30"
                  data-testid="workExperienceCurrent"
                  id="input"
                  name="workExperienceCurrent"
                  placeholder="For example 2"
                  required=""
                  type="number"
                />
              </label>
            </div>
          </div>
        </div>
        <button
          class="_button_148529 _form__button_b45c30"
          data-testid="submit"
          type="submit"
        >
          Continue
        </button>
      </form>
    </div>
  </div>
</div>
`;
